restore/restore_internal_test.go

```diff
@@ -227,4 +236,142 @@ var _ = Describe("restore internal tests", func() {
+			Expect(err).ToNot(HaveOccurred())
+
+			err = os.WriteFile(path.Join(os.TempDir(), "backup/gpseg-1/backups/20250815/20250815120713/gpbackup_20250815120713_statistics.sql"), []byte(
+				`UPDATE pg_class
+SET
+        relpages = 194::int,
+        reltuples = 10000.000000::real
+WHERE oid = 'public.t1'::regclass::oid;
+
+DELETE FROM pg_statistic WHERE (starelid, staattnum) =
+        (SELECT attrelid, attnum FROM pg_attribute WHERE attrelid = 'public.t1'::regclass::oid AND attname = 'a');
+INSERT INTO pg_statistic SELECT
+        attrelid,
+        attnum,
+        false::boolean,
+        0.000000::real,
+        8::integer,
+        -1.000000::real,
+        2::smallint,
+        3::smallint,
+        0::smallint,
+        0::smallint,
+        0::smallint,
+        412::oid,
+        412::oid,
+        0::oid,
+        0::oid,
+        0::oid,
+        NULL::real[],
+        '{"0.313703984"}'::real[],
+        NULL::real[],
+        NULL::real[],
+        NULL::real[],
+        array_in('{"1","100",}', 'int8'::regtype::oid, -1),
+        NULL,
+        NULL,
+        NULL,
+        NULL
+FROM pg_attribute WHERE attrelid = 'public.t1'::regclass::oid AND attname = 'a';
+`), 0777)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = os.WriteFile(path.Join(os.TempDir(), "backup/gpseg-1/backups/20250815/20250815120713/gpbackup_20250815120713_toc.yaml"), []byte(`
+statisticsentries:
+- schema: public
+  name: t1
+  objecttype: STATISTICS
+  referenceobject: ""
+  startbyte: 0
+  endbyte: 129
+  tier:
+  - 0
+  - 0
+- schema: public
+  name: t1
+  objecttype: STATISTICS
+  referenceobject: ""
+  startbyte: 129
+  endbyte: 299
+  tier:
+  - 0
+  - 0
+- schema: public
+  name: t1
+  objecttype: STATISTICS
+  referenceobject: ""
+  startbyte: 299
+  endbyte: 966
+  tier:
+  - 0
+  - 0
+`), 0777)
+
+			gplog.InitializeLogging("gprestore", path.Join(os.TempDir(), "gprestore_test"))
+			gplog.SetLogFileVerbosity(gplog.LOGERROR)
+
+			opts = &options.Options{}
+
+			globalTOC = toc.NewTOC(path.Join(os.TempDir(), "/backup/gpseg-1/backups/20250815/20250815120713/gpbackup_20250815120713_toc.yaml"))
+			globalTOC.InitializeMetadataEntryMap()
+
+			DeferCleanup(func() {
+				err := os.RemoveAll(path.Join(os.TempDir(), "backup"))
+				Expect(err).NotTo(HaveOccurred())
+			})
+		})
+
+		BeforeEach(func() {
+			configCoordinator := cluster.SegConfig{ContentID: -1, Hostname: "localhost", DataDir: "gpseg-1"}
+			testCluster := cluster.NewCluster([]cluster.SegConfig{configCoordinator})
+			testFPInfo := filepath.NewFilePathInfo(testCluster, "", "20250815120713",
+				"gpseg", false)
+			testFPInfo.SegDirMap[-1] = path.Join(os.TempDir(), "/backup/gpseg-1")
+			SetFPInfo(testFPInfo)
+
+			connectionPool, mock, _, _, _ = testhelper.SetupTestEnvironment()
+			connectionPool.Version = dbconn.NewVersion("6.0.0")
+		})
+
+		DescribeTable("Restore statistic with different backup versions", func(backupVersion string, needNestedLoop bool) {
```

Comment:
Maybe also add test to ensure we fatally error with a helpful message if version doesn’t match the pattern?

Reply:
done